sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataStreamReaderWriterSuite.scala

```diff
@@ -815,3 +815,26 @@ class DataStreamReaderWriterSuite extends StreamTest with BeforeAndAfter {
     }
   }
 }
+
+class DataStreamTableAPISuite extends StreamTest with BeforeAndAfter {
+  test("table API") {
```

Comment:
Looks like tests are not sufficient. Please take a look at the cases I covered in my PR: 
https://github.com/apache/spark/pull/29767/files#diff-20df0514e33057039a03be7442a439c1R822-R948

Reply:
Thanks, as your suggestion, let's go through the writer PR first.